lab-michael/routes/recipe-routes.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const storage = require('../lib/storage.js');
+const response = require('../lib/response.js');
+const Recipe = require('../model/recipe.js');
+
+module.exports = function(router) {
+  router.get('/api/recipe', function(req, res){
+    if(req.url.query.id){
+      storage.fetchItem('recipe', req.url.query.id)
+      .then(recipe => {
+        res.writeHead(200, {
+          'Content-Type': 'application/json',
+        });
+        res.write(JSON.stringify(recipe));
+        res.end();
+      })
+      .catch(err => {
+        console.error(err);
+        res.writeHead(404, {
+          'Content-Type': 'text/plain',
+        });
+        res.write('not found\n');
+        res.end();
+      });
+      return;
+    } else {
+      if(!req.url.query.id){
+        storage.fetchAll('recipe', req.url.query.id)
```

Comment:
Why are you passing in an id (which shouldn't exist at this level) to the `fetchAll` method? Does it take two arguments?